pkg/frontend/v2/frontend_test.go

```diff
@@ -979,6 +979,66 @@ func TestFrontend_Protobuf_ReadingCancelledRequest(t *testing.T) {
 	require.Nil(t, msg)
 }
 
+func TestFrontend_Protobuf_ReadingCancelledRequestAfterResponseReceivedFromQuerier(t *testing.T) {
```

Comment:
This is a surprisingly straightforward test for now subtle this issue is :partying_face: